pyglossary/plugins/appledict_bin.py

```diff
@@ -154,6 +151,36 @@ def open(self, filename):
 			f"number of entries: {self._wordCount}"
 		)
 
+	def extract_metadata(self, infoplist_filepath: str):
+		if not isfile(infoplist_filepath):
+			raise IOError(
+				f"Could not find 'Info.plist' file, "
+				"Please provide 'Contents/' folder of the dictionary that contains"
+			)
+		dict_metadata: Dict
+		import biplist
+		try:
+			dict_metadata = biplist.readPlist(infoplist_filepath)
+			print(dict_metadata.get('CFBundleDisplayName'))
+		except (biplist.InvalidPlistException, biplist.NotBinaryPlistException):
+			try:
+				import plistlib
+				dict_metadata = plistlib.loads(open(infoplist_filepath, 'rb').read())
+			except Exception as e:
+				raise IOError(
+					"'Info.plist' file is malformed, "
+					f"Please provide 'Contents/' with a correct 'Info.plist'. {e}"
+				)
+		self._glos.setInfo('name', dict_metadata.get('CFBundleDisplayName'))
+		self._glos.setInfo('copyright', dict_metadata.get('DCSDictionaryCopyright'))
+		self._glos.setInfo('author', dict_metadata.get('DCSDictionaryManufacturerName'))
+		self._glos.setInfo('edition', dict_metadata.get('IDXDictionaryVersion'))
+		self._glos.setInfo('CFBundleIdentifier', dict_metadata.get('CFBundleIdentifier'))
+		if 'DCSDictionaryLanguages' in dict_metadata:
+			dict_metadata_langs = dict_metadata.get('DCSDictionaryLanguages')[0]
+			self._glos.setInfo('sourceLang', dict_metadata_langs['DCSDictionaryDescriptionLanguage'])
```

Comment:
We should extract language code ("en" from "en_GB"), and then set `glos.sourceLangName`. Same for targetLang.